### DISCUSSION TITLE
Possible to create new Project to read an digital display?

### DISCUSSION THREAD

**Discussion by PanicMan** [jomjol/AI-on-the-edge-device#1465] [General]
### The Feature

Hello,
I have an Heatmeter which has an digital display which shows every 4sec the current kWh value on the display.
![IMG_20221201_180740](https://user-images.githubusercontent.com/4090506/205116407-567ede60-639e-49a1-8e98-313a844048af.jpg)
Is it possible to create an fork of that project which notices when the value is shown and then check the digits and transmits it to mqtt server?
I know that there is an optical interface, but that is disabled by the vendor.
Thanks a lot! 
Greetings Jack

**Comment by ND1988** [comment]
+1

**Comment by caco3** [comment]
Sure you can create a fork and create your own version!

To be able to catch the time when the value is shown, you would have to switch from taking periodic images to taking a video and process it in real time. I am pretty sure the ESP32 does not have the required performance for this!

I think the best you can do is to run on a fast interval, eg. 3 minutes and accept that you only get once every 12 minutes a goo value.

Another option would be to disable all rate validations and postprocess the values on an external system. That system then would need to separate the good from the empty readings which would not be too difficult.

**Reply by PanicMan** [reply]
Yeah, I have also an RPi4 running OH3 and doing all the time nearly nothing...
But I think its not a big deal, at least one value in a minute would be enough... So just trying to get an value every 3 seconds an hoping to catch at least one value per minute... Sure I can make an Fork and I'm also able to program c++, so, just as a hint, where should I start?

**Comment by ND1988** [comment]
At least for me the 1 minute interval is working, so this reduces the waiting time for a synchronized shot.

What would be helpful, is a mode which is meant for blinking displays.
Here maybe it is possible to select the blinking interval, and the software suggests an picture taking interval.
Also there can be a button in this mode, to start the picture taking synchronized with the blinking (push button = first picture will be taken)
Open topic then will be reboots. Maybe a stored value, when the first picture has been taken and then a calculation when the next correct time for the next picture is, can solve this issue.

**Reply by caco3** [reply]
I understand your idea, how ever it would be quite some effort and we do not have the manpower do do it atm.

**Comment by maurcabezas** [comment]
Hello, Actually I have a Digital LCD power meter, which is blinking each 8 second between two tariffs.
I am using Home Assistant and MQTT, As I do not really need to consumption every 3 minutes, I set as interval time 12.39 minutes. Always you have to set an ODD amount of period (if you chose even, always you will have the same value, I do not your case but I need to read both values) , as @caco3 wrote, you have to disable almost all the native post processing and use, for example homeassistant with template sensor, then my setup of the sensor in config.yaml HA is:


```
mqtt:
  sensor:
    - name: "Main Power Meter"
      state_topic: "powermeter/main/value"
      value_template: |-
        {%  if value != "unknown" %}
        {{ value | int}}
        {%else%}
        {{states('sensor.main_power_meter')}}
        {% endif %} 
      state_class: total_increasing 
      unit_of_measurement: "kWh"
      icon: mdi:flash
      device_class: energy

template:
    - trigger:
        - platform: time_pattern
          hours: "0"
          minutes: "0"
    - sensor:
      - name: "Main Power Meter T1"
        unit_of_measurement: "kWh"
        state_class: total_increasing
        device_class: energy
        state: >
          {%if (states('sensor.main_power_meter') | int /10000) | int == 2 
          and (states('sensor.main_power_meter') | int) >= states('sensor.last_t1') | int 
          and (states('sensor.main_power_meter') | int) <= states('sensor.last_t1') | int + 5 %}
          {{states('sensor.main_power_meter')}}
          {% else %}
          {{ states('sensor.last_t1') }}
          {% endif %}

    - trigger:
        - platform: time_pattern
          hours: "0"
          minutes: "0"
    - sensor:
      - name: "Main Power Meter T2"
        unit_of_measurement: "kWh"
        state_class: total_increasing
        device_class: energy
        state: >
          {%if (states('sensor.main_power_meter') | int /10000) | int == 6 
          and (states('sensor.main_power_meter') | int) >= states('sensor.last_t2') | int 
          and (states('sensor.main_power_meter') | int) <= states('sensor.last_t2') | int + 10 %}
          {{states('sensor.main_power_meter')}}
          {% else %}
          {{ states('sensor.last_t2') }}
          {% endif %}
```
I have T1 and T2 as tariff 1 and 2 respectively, T1 is around 20K and, T2 around 60K, the number from MQTT is sensor.main_power_meter,  in case of error in reading number, I accept only numbers equals or greater than T1 or T2 for each case and.. no more then 5 or 10. Also to avoit 'unknown' or 'unavailable' values after rebooting HA, I set another sensor as "sensor.last_T2' which I take it by quering the database with mysql.

if you are planning to use it, please write me!

Cheers!

**Reply by caco3** [reply]
my meter also shows a T1 resp. T2.
idea: you could add this as a 2nd Number so HA can pick the right tariff.

**Reply by maurcabezas** [reply]
You mean, read two numbers..? but how the system (without HA template) will known that T2 as to be, for example 60K and,m T1~ 20K? could you share your conf?
Thank you.

**Reply by PanicMan** [reply]
What sensor are you reading out here? That is an energy meter sensor, right? We need something what can recognize the digits on an digital display. Every readout over the optical interface is not working as that is turned off by vendor or supplier... :(

**Reply by maurcabezas** [reply]
Yes.. is a energy meter , I read with "AI on the edge" the digital display 
![IMG_20221206_144128](https://user-images.githubusercontent.com/8492276/205927848-b435e114-bfcb-4ebd-be77-60f7193c596e.jpg)

**Reply by PanicMan** [reply]
Ok, great, and how exactly do you read the display?

**Reply by caco3** [reply]
> You mean, read two numbers..? but how the system (without HA template) will known that T2 as to be, for example 60K and,m T1~ 20K? could you share your conf? Thank you.

I do not have any config as i dont use it for the energy meter.

My Idea was: If it either displays T1 or T2, you could digitalize the 1 resp. 2 and use it as a separate sensor for better differentiation in the template sensor.
That would be more stable than your current approach of simply comparing to the previous value. That will start causing issues once the 2 tariffs start providing values that are close to each other.

**Comment by sheminasalam** [comment]
I have also been looking into how I could do this. I tried to use the [seven segment lcd display integration](https://www.home-assistant.io/integrations/seven_segments/) of Home Assitant instead of using the esp32 to process the images but i failed get the image processing working in HA. Now this would be better. Thanks. Please do update more details if any.
